lesson5/src/main/java/com/andersenlab/util/Validator.java

```diff
@@ -0,0 +1,59 @@
+package com.andersenlab.util;
+
+import com.andersenlab.entity.BusTicket;
+import com.andersenlab.entity.TicketType;
+
+import java.time.LocalDate;
+import java.util.EnumSet;
+import java.util.List;
+
+public class Validator {
+    public List<BusTicket> validateStartDateExistsInRequiredTypes(List<BusTicket> list) {
+        EnumSet<TicketType> requiredDataTypes = EnumSet.of(TicketType.DAY, TicketType.WEEK, TicketType.YEAR);
+        EnumSet<TicketType> unrequiredDataTypes = EnumSet.of(TicketType.MONTH);
+        return list.stream()
+                .filter(ticket -> (requiredDataTypes.contains(ticket.getTicketType()) &&
+                        ticket.getStartDate() != null && !ticket.getStartDate().isBlank()) ||
+                        unrequiredDataTypes.contains(ticket.getTicketType()))
+                .toList();
+    }
+
+    public List<BusTicket> validatePriceExisting(List<BusTicket> list) {
+        return list.stream()
+                .filter(ticket -> !priceZeroNullOrBlank(ticket.getPrice()))
+                .toList();
+    }
+
+    public List<BusTicket> validateStartDateNotInFuture(List<BusTicket> list) {
+        return list.stream()
+                .filter(busTicket -> stringNotNullOrBlank(busTicket.getStartDate()))
```

Comment:
no sense of having two filters right after each other, just combine their predicates